uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrls.spec.js

```diff
@@ -1288,6 +1288,137 @@
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.more_information_required,
+            status.approved
+          ]);
+        });
+
+        it('returns *More Information Required, Approve, Reject, Cancel* when previous status is *Awaiting Approval*', function () {
+          $ctrl.request.status_id = status.awaiting_approval.value;
+
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.more_information_required,
+            status.approved,
+            status.rejected,
+            status.cancelled
+          ]);
+        });
+
+        it('returns *Approve, More Information Required, Reject, Cancel* when previous status is *More Information Required*', function () {
+          $ctrl.request.status_id = status.more_information_required.value;
+
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.more_information_required,
+            status.approved,
+            status.rejected,
+            status.cancelled
+          ]);
+        });
+
+        it('returns *Approve, More Information Required, Reject, Cancel* when previous status is *Rejected*', function () {
+          $ctrl.request.status_id = status.rejected.value;
+
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.more_information_required,
+            status.approved,
+            status.rejected,
+            status.cancelled
+          ]);
+        });
+
+        it('returns *Approve, More Information Required, Reject, Cancel* when previous status is *Approved*', function () {
+          $ctrl.request.status_id = status.approved.value;
+
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.more_information_required,
+            status.approved,
+            status.rejected,
+            status.cancelled
+          ]);
+        });
+
+        it('returns *Awaiting Approval , Approve, More Information Required, Reject, Cancel* when previous status is *Cancelled*', function () {
+          $ctrl.request.status_id = status.cancelled.value;
+
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
+            status.awaiting_approval,
+            status.more_information_required,
+            status.approved,
+            status.rejected,
+            status.cancelled
+          ]);
+        });
+
+        function expectStatusToBeEqual (array1, array2) {
+          expect(_.isArray(array1)).toBe(true, array1, 'is not an array');
+
+          expect(array1.length).toBe(
+            array2.length,
+            array1.map((e) => e.name), array2.map((e) => e.name)
+          );
+
+          array1.forEach(function (element) {
+            expect(array2.indexOf(element) >= 0).toBe(
```

Comment:
This was only on the unit test since the runner browser is Chrome and it supports arrow functions by default.

I'll remove them though.